### PR TITLE
feat: vscode devcontainer configuration

### LINKED ISSUE #371: vscode devcontainer support?
How about adding config for a vscode devcontainer for the project? devcontainers are friggin awesome

no changes are needed to the codebase, just some config files

See https://github.com/ludeeus/integration_blueprint

### PR DESCRIPTION
**Pull request by bj00rn**
Add vscode devcontainer configuration.

fixes #371

### CODE REVIEW

**Comment by bj00rn** [comment]
@argoyle not sure why hassfest is complaining, seems to be unlelated to this PR

**Comment by argoyle** [comment]
New requirements on services perhaps? I'll take a look at the PR tomorrow and see if I can find anything related to those errors.

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+Visual Studio code will ask "Reopen in Container", this will start the build of the container.
+
+_If you don't see this notification, open the command palette and select `Remote-Containers: Reopen Folder in Container`._
+
+### Start devcontainer
+
+Open the command palette and select `Tasks: Run Task` then select `Run Home Assistant on port 8123` this will start the Home Assistant instance.
+
+The devcontainer canbe restarted by opening the command palette and selecting `Tasks: Restart Running Task`, then select the task `Run Home Assistant on port 8123`.
```
Comment (argoyle) [inline .devcontainer/README.md:31]:
```suggestion
The devcontainer can be restarted by opening the command palette and selecting `Tasks: Restart Running Task`, then select the task `Run Home Assistant on port 8123`.
```

File: .devcontainer/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+When the devcontainer is running, open a browser and navigate to `localhost:8123` and finish configuration of the instance.
```
Comment (argoyle) [inline .devcontainer/README.md:33]:
```suggestion
When the devcontainer is running, open a browser and navigate to `http://localhost:8123` and finish configuration of the instance.
```

**Review by argoyle — CHANGES_REQUESTED** [review]
Just a small typo and an addition of http:// for extra clarity otherwise 💯

**Comment by argoyle** [comment]
> @argoyle not sure why hassfest is complaining, seems to be unlelated to this PR

I've added name to the services so just rebase your branch and hassfest should be happy again I think.

**Comment by bj00rn** [comment]
@argoyle are you using any formatter for this project? black or the likes?

vscode can be quite intrusive with formatting if user settings differ from the code style of the project, unless formatter is set in .vscode/settings.json

**Comment by argoyle** [comment]
Not at the moment. Far from a Python ninja so not well versed in the ecosystem and best practices. Not against adding it to make it easier to contribute changes without adding mixed code style etc. Let's merge this as is now and add more stuff later.

**Comment by bj00rn** [comment]
> Not at the moment. Far from a Python ninja so not well versed in the ecosystem and best practices. Not against adding it to make it easier to contribute changes without adding mixed code style etc. Let's merge this as is now and add more stuff later.

Made a PR to enforce codestyle using black and isort. Have a look, the most notable change is double qoutes and line width

**Comment by bj00rn** [comment]
https://github.com/henricm/ha-ferroamp/pull/375

**Pull request merged**